search/reverse_geocoder.cpp

```diff
@@ -98,6 +117,44 @@ size_t ReverseGeocoder::GetMatchedStreetIndex(string const & keyName,
     }
   }
 
-  return (result < count ? result : streets.size());
+  return result;
 }
+
+void ReverseGeocoder::GetNearbyAddress(m2::PointD const & center,
+                                       Building & building, Street & street)
+{
+  vector<Building> buildings;
+  GetNearbyBuildings(center, buildings);
+
+  vector<Street> streets;
+  unique_ptr<search::v2::HouseToStreetTable> table;
+  MwmSet::MwmId mwmId;
+
+  for (auto const & b : buildings)
+  {
+    if (!table || mwmId != b.m_id.m_mwmId)
+    {
+      auto handle = m_index.GetMwmHandleById(b.m_id.m_mwmId);
```

Comment:
It's not correct to take handle here and return it at the end of the code block, since mwm may be deleted.